modules/claims_api/spec/lib/claims_api/error/error_handler_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'claims_api/special_issue_mappers/bgs'
+require 'token_validation/v2/client'
+require 'claims_api/error/error_handler'
+# require 'spec_helper'
+
+describe ApplicationController, type: :controller do # rubocop:disable RSpec/MultipleDescribes
+  let(:scopes) { %w[claim.write] }
+
+  controller do
+    skip_before_action :authenticate
+
+    def test_invalid_token
+      raise ::Common::Exceptions::TokenValidationError
+    end
+  end
+
+  before do
+    routes.draw do
+      get 'test_invalid_token' => 'anonymous#test_invalid_token'
+    end
+  end
+
+  it 'catches an invalid token' do
+    with_okta_user(scopes) do |auth_header|
+      # Following the normal headers: auth_header pattern fails due to
+      # this rspec bug: https://github.com/rspec/rspec-rails/issues/1655
+      # This is the recommended workaround from that issue thread.
+      request.headers.merge!(auth_header)
+
+      get :test_invalid_token
+
+      expect(response).to have_http_status(:unauthorized)
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body['errors'].size).to eq(1)
+      expect(parsed_body['errors'][0]['title']).to eq('Token Validation Error')
+      expect(parsed_body['errors'][0]['detail']).to eq('Token Validation Error')
+    end
+  end
+end
```

Comment:
This works as intended now. I was unable to get a similar test working, because creating a `test_expired_token` route without skipping `authenticate` hits that named method [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/concerns/authentication_and_sso_concerns.rb#L16) instead of [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/openid_application_controller.rb#L22). If you can figure out how to get it to call the right inherited method, my updated approach below should work for that, too. Otherwise, it's solved below (see other comment)